components/net/tests/main.rs

```diff
@@ -95,6 +97,76 @@ fn create_embedder_proxy() -> EmbedderProxy {
     }
 }
 
+fn create_embedder_proxy_and_receiver() -> (EmbedderProxy, EmbedderReceiver) {
```

Comment:
this has some duplication with the function right above called `create_embedder_proxy`. I decided not to merge them but I'm open to doing so if you think it's better that way.